libraries/libfc/include/fc/crypto/bls_common.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <fc/crypto/common.hpp>
+
+namespace fc::crypto::blslib {
+
+   template <typename Container>
+   static Container serialize_base58(const std::string& data_str)
+   {
+
+      using wrapper = checksummed_data<Container>;
+
+      wrapper wrapped;
+
+      auto bin = fc::from_base58(data_str);
+      fc::datastream<const char*> unpacker(bin.data(), bin.size());
+      fc::raw::unpack(unpacker, wrapped);
+      FC_ASSERT(!unpacker.remaining(), "decoded base58 length too long");
+      auto checksum = wrapper::calculate_checksum(wrapped.data, nullptr);
+      FC_ASSERT(checksum == wrapped.check);
+
+      return wrapped.data;
+   }
+
+   template <typename Container>
+   static std::string deserialize_base58( Container data, const yield_function_t& yield) {
+
+      using wrapper = checksummed_data<Container>;
+      
+      wrapper wrapped;
+
+      wrapped.data = data;
+      yield();
+      wrapped.check = wrapper::calculate_checksum(wrapped.data, nullptr);
+      yield();
+      auto packed = raw::pack( wrapped );
+      yield();
+      auto data_str = to_base58( packed.data(), packed.size(), yield );
+      yield();

```

Comment:
I followed the same pattern than here : https://github.com/AntelopeIO/leap/blob/main/libraries/libfc/include/fc/crypto/common.hpp#L98

But I agree that, at least for the last yield(), it seems unecessary.

Reply:
My guess is those `yield()`s are there in common.hpp so that string/json to bin doesn't get out of control some how on API calls that deal with ABI serialization? Not sure if this concern applies to BLS types or not since, while I'm a little out of the loop now, I didn't think we were wiring them up as ABI types?

Reply:
I can imagine that we are worried about an attach using a well crafted string which would cause `to_base58()` to take excessively long time, but it is not clear to me what the other `yield()` protect against.

Reply:
We are not handling them as ABI types; we decided to pass them as strings. Deserialization of these types I believe will all be in initialization of `producer_plugin` or leap util. Likely best to keep the `yield()` calls incase the use is expanded in the future. If we log them somewhere there is the potential for the logger to pause the application.